tmt/steps/provision/testcloud.py

```diff
@@ -556,8 +556,8 @@ def start(self):
 
         # Boot the virtual machine
         self.info('progress', 'booting...', 'cyan')
-        self._instance.ram = self.memory
-        self._instance.disk_size = self.disk
+        self._instance.ram = int(self.memory)
+        self._instance.disk_size = int(self.disk)
```

Comment:
This is too late, both `memory` and `disk` come from `TestcloudGuestData` and are defined as `int`, and if they are not at this point, it's a bug. We need to fix the place where code took `memory` and assigned it to data `memory` attribute without making sure it is indeed an integer.

One candidate would be https://github.com/teemtee/tmt/pull/1446/files#diff-06f5216200e1c8ddc7476f7e6a34f74600098c33344a4e7fc3a21ff5cfce5985R281, the other https://github.com/teemtee/tmt/pull/1446/files#diff-06f5216200e1c8ddc7476f7e6a34f74600098c33344a4e7fc3a21ff5cfce5985R264. I must admit I don't know what the relationship between `go()` and `wake()` is, who runs first and who's therefore responsible for `memory` and `disk` being converted to integers. I'd say `go()` is the right place, since `wake()` is not creating the instance of `TestcloudGuestData`, just consuming it, and that instance should be correct at that point. So it'd be up to `go()` to create valid `data` package.

Reply:
Well, this fixed the issue for me so I don't understand why it is too late...

Reply:
Because you are leaving one bug left. At this point in code flow, `memory` is expected to be `int`, at least according to type annotations and mypy magic. If it's not an integer, then the question is why. What happened that `memory: int` declaration has not been violated? Yes, you fix your use case, and the plugin would move on, but the bug is still present: something took an input, a string, and assigned it to a field the code believes is an integer. This bug would remain.

The real issue is not `self.memory` being a string, it's rather `self.memory` being a string *when it's supposed to be an integer* - either string is also acceptable here, but then `int(self.memory)` must be complemented by `memory: Union[str, int]` change to `TestcloudGuestData`, to make this acceptance formally known, or string is not acceptable here, `TestcloudGuestData` is correct, and then `int(self.memory)` **at this place** is not the fix.

Reply:
In other words, right move, wrong place. I for one would strongly advise against allowing `Union[str, int]` - if `memory` is supposed to be an integer, then there's no reason to leave `str` in play. The place where value enters tmt's internals should be responsible for conversion, on the boundary between outside and tmt, and right I'd say the bug's exactly there.